File: python/problem3.py

def factors(number):
    factors = []

    n = 2
    m = number / 2

    while n < m:
        if number % n == 0:
            factors.append(n)

        n += 1

    return factors

def primefactors(number):
    remaining = number
    factors = set()

    n = 2
    m = number / 2

    while n < m:
        while remaining % n == 0:
            factors.add(n)
            remaining = remaining / n

        if n > 2:
            n += 2
        else:
            n += 1

        if n > remaining:
            break

    factors = list(factors)
    factors.sort()

    return factors

#print primefactors(13195)
print primefactors(600851475143)

File: swift/problem3.swift
func primefactors(number: UInt) -> Array<UInt>
{
    var n: UInt = 2
    var remaining: UInt = number
    var factors = Set<UInt>()

    while n <= remaining {
        while remaining % n == 0 {
            factors.insert(n)
            remaining = remaining / n
        }

        n += (n > 2 ? 2 : 1)
    }

    return factors.sort()
}

print(primefactors(600851475143))
